Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSFitShiftScale.py

```diff
@@ -144,8 +144,17 @@ def _determine_factors(self, q_high_angle, q_low_angle, mode, scale, shift):
                                                                                          front_data=q_high_angle,
                                                                                          q_min=q_min, q_max=q_max)
 
-        fit = self.createChildAlgorithm('Fit')
+        # The front_data_corrected data set is used as the fit model. Setting the IgnoreInvalidData on the Fit algorithm
+        # will not have any ignore Nans in the model, but only in the data. Hence this will lead to unreadable
+        # error messages of the fit algorithm. We need to catch this before the algorithm starts
+        y_model = front_data_corrected.dataY(0)
+        y_data = rear_data_corrected.dataY(0)
+        if any([np.isnan(element) for element in y_model]) or any([np.isnan(element) for element in y_data]):
+            raise RuntimeError("Trying to merge two merge the two reduced data sets for HAB and LAB failed. "
```

Comment:
"Trying to merge two merge the two reduced ..." doesn't make sense to me. Do you mean "Trying to merge the two reduced ..."?